projects/RabbitMQ.Client/client/impl/Connection.cs

```diff
@@ -74,12 +73,17 @@ public Connection(ConnectionConfig config, IFrameHandler frameHandler)
             _session0 = new MainSession(this);
             _model0 = new Model(_config, _session0);;
 
-            ClientProperties = new Dictionary<string, object?>(_config.ClientProperties)
+            ClientProperties = new Dictionary<string, object?>
```

Comment:
Why changing this? (deoptimizes the creation of the dictionary)

Reply:
I'd like the data structures used in `ConnectionConfig` to be immutable.

Reply:
It has nothing to do with immutability. [The constructor](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.-ctor?view=net-6.0#system-collections-generic-dictionary-2-ctor(system-collections-generic-ienumerable((system-collections-generic-keyvaluepair((-0-1)))))) makes a copy of the dictionary provided. You just fill it manually which is way less efficient.

Reply:
> It has nothing to do with immutability.

Sure it does. I changed `ClientProperties` in `ConnectionConfig` to be `IReadOnlyDictionary<>`. The `Dictionary<>` constructor does not take that type as an argument, which is why I had to change it here.

Reply:
The one linked above is `Dictionary<TKey,TValue>(IEnumerable<KeyValuePair<TKey,TValue>>)` which is compatible with `IReadOnlyDictionary`

Reply:
I'll re-check but I'm pretty sure I got a compilation error when I tried to do something like that 🤷

Reply:
```
 C:\Users\bakkenl\development\rabbitmq\rabbitmq-dotnet-client\projects\RabbitMQ.Client\client\impl\Connection.cs(76,64): error CS1503: Argument 1: cannot convert from 'System.Collections.Generic.IReadOnlyDictionary<string, object?>' to 'System.Collections.Generic.IDictionary<string, object?>' [C:\Users\bakkenl\development\rabbitmq\rabbitmq-dotnet-client\projects\RabbitMQ.Client\RabbitMQ.Client.csproj]
```

Reply:
Hmm sorry, I've checked against .NET Core 3.1, not NetStandard2.0, which doesn't have it true.

Reply:
We could #ifdef it for .NetStandard2.0

Reply:
Well, `main` is going to target `net6.0` before it's released as version `7.0` so I'll punt this. Thanks for the note!